### PR TITLE
CLN: Address some TODO comments

### LINKED ISSUE #23098: CLN: commented-out code in TimedeltaIndex._partial_td_slice
AFAIK this is the only dead code in the code-base

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #23098
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

I am technically moving some public (non top level) helper functions and making them private, but I don't think it warrants a depreciation.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8040,3 +8040,13 @@ def _from_nested_dict(data):
 
 def _put_str(s, space):
     return u'{s}'.format(s=s)[:space].ljust(space)
+
+
```
Comment (jreback) [inline pandas/core/frame.py]: don't put this here, pandas/core/indexing.py is better or pandas/core/index/api.py

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/frame.py]: `_get_info_slice` is only used once in `frame.py` and nowhere else. I was thinking we might as well avoid the import.

Comment (jreback) [reply pandas/core/frame.py]: ok, can you just inline it then where it is used (make it a local function) instead.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -3205,7 +3205,7 @@ def is_dtype_instance_mapper(idx, dtype):
                 exclude_these.iloc[idx] = not any(map(f, exclude))
 
         dtype_indexer = include_these & exclude_these
-        return self.loc[com.get_info_slice(self, dtype_indexer)]
```
Comment (jreback) [inline pandas/core/frame.py]: just move the local function here

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**